protobuf/lib/src/protobuf/coded_buffer_writer.dart

```diff
@@ -100,8 +100,9 @@ class CodedBufferWriter {
     final wireFormat = _wireTypes[_valueTypeIndex(valueType)];
 
     if ((fieldType & PbFieldType._REPEATED_BIT) != 0) {
-      for (var i = 0; i < fieldValue.length; i++) {
-        _writeValue(fieldNumber, valueType, fieldValue[i], wireFormat);
+      List list = fieldValue;
```

Comment:
(If we know that those lists are actually a specific list (e.g. `PbList`) it may help to strengthen the type further)

Reply:
This value is `List` in some cases: https://github.com/google/protobuf.dart/blob/29f17490bcfe5e19d77482544c82a559df88d9b9/protobuf/lib/src/protobuf/unknown_field_set.dart#L207

We could probably use `PbList` in that code. I'll do that in another PR.